modules/formulize/include/formdisplay.php

```diff
@@ -175,21 +175,50 @@ function _drawValidationJS($skipConditionalCheck) {
 		
 		$elements = $this->getElements( true );
 		foreach ( $elements as $elt ) {
-			if ( method_exists( $elt, 'renderValidationJS' ) ) {
-				if(substr($elt->getName(),0,3)=="de_" AND !$skipConditionalCheck) {
-					$checkConditionalRow = true;
-				} else {
-					$checkConditionalRow = false;
-				}
-				$js = $elt->renderValidationJS();
-				if($js AND $checkConditionalRow) {
-					$fullJs .= "if(window.document.getElementById('formulize-".$elt->getName()."').style.display != 'none') {\n".$js."\n}\n\n";
-				} elseif($js) {
-					$fullJs .= "\n".$js."\n";
-				}
-			}
-		}
-		
+            if (method_exists($elt, 'renderValidationJS')) {
+                if (substr($elt->getName(), 0, 3) == "de_" AND !$skipConditionalCheck) {
+                    $checkConditionalRow = true;
+                } else {
+                    $checkConditionalRow = false;
+                }
+                $js = $elt->renderValidationJS();
+                if ($js AND $checkConditionalRow) {
+                    $fullJs .= "if(window.document.getElementById('formulize-" . $elt->getName() . "').style.display != 'none') {\n" . $js . "\n}\n\n";
+                } elseif ($js) {
+
+                    $newStr = "";
+
+                    $elementNameParts = explode("if", $js);
+                    //echo '<pre>'; print_r($elementNameParts); echo '</pre>';
+                    for ($x = 1; $x < sizeof($elementNameParts); $x++) {
+                        //subfield elements
+                        $current = $elementNameParts[$x];

```

Comment:
You just need to explode $elt->getName() to get the ID, no need to parse all the javascript and explode on If, etc